src/Symfony/Component/Dotenv/Tests/DotenvTest.php

```diff
@@ -430,4 +430,14 @@ public function testDoNotUsePutenv()
         $this->assertSame('no', $_ENV['TEST_USE_PUTENV']);
         $this->assertFalse(getenv('TEST_USE_PUTENV'));
     }
+
+    public function testGetVariablesValueFromEnvFirst()
+    {
+        $_ENV['APP_ENV'] = 'prod';
+        $dotenv = new Dotenv(true);
+        $test = "APP_ENV=dev\nTEST=foo_\${APP_ENV}";
+        $values = $dotenv->parse($test);
+
+        $this->assertSame('foo_prod', $values['TEST']);
```

Comment:
It should be `foo_dev` no ?

Reply:
No must return the value of $_ENV and not the value of .env file.

when you run the command with **prod**  or other value not the same in .env file the value of TEST is wrong (it is same as in .env file) .

Then I run the same command with prod environment (composer dump-env prod), the value of TEST is wrong (it is same as for dev)

```
// .env.locale.php

return array (
  'APP_ENV' => 'prod',
  'TEST' => 'foo_dev',
  'APP_SECRET' => '2eb810c79fba0dd5c029a2fa53bfdb51',
);
```

Reply:
Ah, yes my bad. I read it too quickly